content/audio.md

```diff
@@ -17,50 +21,98 @@ section: software-troubleshooting
+
+## Verify Audio Devices Detected by the System
+
+Below are set of commands that can be used to detect various information about your audio device.
+
+List all detected audio devices:
+
+```
+aplay -l
+```
+
+List audio input devices:
+
+```
+arecord -l
+```
+
+List the current sound cards:
+
+```
+cat /proc/asound/cards
+```
+
+List kernel modules related to audio:
+
+```
+sudo lspci -k | grep -A 3 -i audio
+```
+
+Your audio devices should appear in these lists.
+
+If not, this may be a driver issue.
+
+## Restart the Audio Services
+
+Restart the audio daemon:
+
+```
+systemctl --user restart wireplumber pipewire pipewire-pulse
+```
+```
+rm -r ~/.config/pulse
+```
+
+This set of commands restarts the <u>PipeWire audio server</u> and its associated services for the current user. This also removes the <u>PulseAudio</u> configuration and will create new default audio configuration files.
 
 ## Check the PulseAudio Controls
 
-The program <u>PulseAudio Volume Control</u> is helpful in figuring out which program is producing audio, where that audio is being routed, what the default input/output devices are, and what the volume levels are set to.  It can be installed using the Pop!\_Shop, or with this command:
+The program <u>PulseAudio Volume Control</u> is helpful in figuring out which program is producing audio, where that audio is being routed, what the default input/output devices are, and what the volume levels are set to. It can be installed using the Pop!_Shop or Cosmic Store:
+
+![audio_3](images/audio/audio_3.png)
+
+or with this command:
 
 ```
 sudo apt install pavucontrol
 ```
 
-Once installed, open the <u>PulseAudio Volume Control</u> app in the Activities or Applications menu, or run `pavucontrol` in a terminal.
+Once installed, open the PulseAudio Volume Control app in the Activities or Applications menu, or run pavucontrol in a terminal.
 
 The "Playback" tab will show a list of applications that are currently playing audio. The volume for each application can be individually controlled.
 
-![Playback tab](/images/audio/pavucontrol-playback.png)
+![audio_4](images/audio/audio_4.png)
 
 Similarly, the "Recording" tab will show a list of applications that are currently recording audio.
 
-![Recording tab](/images/audio/pavucontrol-recording.png)
+![audio_5](images/audio/audio_5.png)
 
 The "Output Devices" tab shows a list of output devices, and an indicator of what's being played out of each device. The green checkmark being selected indicates a device is the default output device.
 
-![Output tab](/images/audio/pavucontrol-output.png)
+![audio_6](images/audio/audio_6.png)
 
 The "Input Devices" tab shows a similar list for input devices.
 
-![Input tab](/images/audio/pavucontrol-input.png)
+![audio_7](images/audio/audio_7.png)
 
 Under "Configuration," each sound card should be listed.
 
-![Configuration tab](/images/audio/pavucontrol-configuration.png)
+![audio_8](images/audio/audio_8.png)
 
 ## Troubleshooting ALSA
```

Comment:
The contents of this section includes two of the same commands you're duplicating earlier in the article (`aplay -l` and `arecord -l`). Do they need to be listed in two different places?